en/docs/develop/setting-up-user-portal-in-a-dev-environment.md

```diff
@@ -0,0 +1,101 @@
+    (https://github.com/wso2/product-is)).
+
+
+### Step 1: Configuring Identity Server
+
+!!! tip
+    Default dev origin, hostname and port set for the User Portal is https://localhost:9000. So the following
+    configurations will have to change accordingly if you change the port or any other configurations.
+
+
+1.  Allowing Cross Origin requests to the Identity Server.
+
+    Cross origin requests will be blocked by default as a security measure. So you have to put the following CORS
+    filter to the config file `<IS_HOME>/repository/resources/conf/templates/repository/conf/tomcat/web.xml.j2` in
+    WSO2 Identity Server distribution pack.
+    ```
+        <filter>
+            <filter-name>CORS</filter-name>
+            <filter-class>com.thetransactioncompany.cors.CORSFilter</filter-class>
+            <init-param>
+                <param-name>cors.allowOrigin</param-name>
+                <param-value>https://localhost:9000</param-value>
+            </init-param>
+            <init-param>
+                <param-name>cors.supportedMethods</param-name>
+                <param-value>GET, HEAD, POST, DELETE, OPTIONS, PATCH, PUT</param-value>
+            </init-param>
+            <init-param>
+                <param-name>cors.exposedHeaders</param-name>
+                <param-value>Location</param-value>
+            </init-param>
+        </filter>
+        <filter-mapping>
+            <filter-name>CORS</filter-name>
+            <url-pattern>/*</url-pattern>
+            <dispatcher>REQUEST</dispatcher>
+            <dispatcher>FORWARD</dispatcher>
+        </filter-mapping>
+    ```
+2.  Add your hostname and port as a trusted FIDO2 origin.
+
+    Whitelist the dev URL in the FIDO configurations found in the file
+    `<IS_HOME>/repository/resources/conf/templates/repository/conf/identity/identity.xml.j2`.
+    ```
+        <FIDO>
+            <WebAuthn>
+                <Enable>{{fido.webauthn.enable}}</Enable>
+            </WebAuthn>
+            <FIDO2TrustedOrigins>
+                    {% for origin in fido.trusted.origins %}
+                    <Origin>{{origin}}</Origin>
+                    {% endfor %}
+                    <Origin>https://localhost:9000</Origin>
+            </FIDO2TrustedOrigins>
+        </FIDO>
+    ```
+
+3.  Restart the Identity Server.
+4.  Configure the callback URLs for the User Portal.
+
+    Log in to the management console `https://<SERVER_HOST>/carbon`
+    Go to service provider listing and click on edit in **User Portal** list item.
+
+    In the edit view, expand the **Inbound Authentication Configuration** section, and further expand the **OAuth/OpenID
+     Connect Configuration**. Click on edit under User Portal list item.
+
+    You can simple change the port to 9000 or you can add a regexp as follows.
+    ```
+        regexp=(https://localhost:9443/user-portal/login|https://localhost:9000/user-portal/login)
+    ```
```

Comment:
Number these as steps indented under step 4. 

1. Log in to the WSO2 IS management console. 

2. Click **Service Providers > List**.

3. Click **Edit** to edit the **User Portal** service provider. 

4. Expand **Inbound Authentication Configuration**, and then expand **OAuth/OpenID Connect Configuration**. Click **Edit**. 

5. Change the **Callback URL** field to reflect the port as 9000 or you can add a regexp as follows.